File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;

        while (true) {
            int[] newArr = new int[arr.length];
            boolean isChanged = false;

            for (int i = 0; i < arr.length; i++) {
                if (arr[i] >= 50 && arr[i] % 2 == 0) {
                    newArr[i] = arr[i] / 2;
                } else if (arr[i] < 50 && arr[i] % 2 == 1) {
                    newArr[i] = arr[i] * 2 + 1;
                } else {
                    newArr[i] = arr[i];
                }
                if (newArr[i] != arr[i]) {
                    isChanged = true;
                }
            }

            if (!isChanged) {
                break;
            }
            arr = newArr;
            answer++;
        }

        return answer;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let answer = 0;
    let prev = [];
    
    while(true){
    let temp = [];
        
    for(let i=0; i<arr.length; i++){
        let value = arr[i];
       if(value >=50 && value % 2 === 0){
           value /=2;
       }else if(value <50 && value % 2 === 1){
            value = value*2 +1;
        }
        temp.push(value);
    }
    
        if(JSON.stringify(arr) === JSON.stringify(temp)){
            break;
        }
        
        arr = temp;
        answer++;
    }
    return answer;
}